docs/proposals/cluster-deployment.md

```diff
@@ -0,0 +1,190 @@
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/autoscaling.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Objective
+
+Simplify the cluster provisioning process for a cluster with one master and multiple worker nodes.
+It should be secured with SSL and have all the default add-ons. There should not be significant 
+differences in the provisioning process across deployment targets (cloud provider + OS distribution)
+once machines meet the node specification.
+
+## Overview
+
+Cluster provisioning can be broken into a number of phases, each with their own exit criteria.
+In some cases, multiple phases will be combined together to more seamlessly automate the cluster setup,
+but in all cases the phases can be run sequentially to provision a functional cluster.
+
+It is possible that for some platforms we will provide an optimized flow that combines some of the steps
+together, but that is out of scope of this document.
+
+## Deployment flow
+
+**Note**: _Exit critieria_ in the following sections are not intended to list all tests that should pass,
+rather list those that must pass.
+
+### Step 1: Provision cluster
+
+**Objective**: Create a set of machines (master + nodes) where we will deploy Kubernetes.
+
+For this phase to be completed successfully, the following requirements must be completed for all nodes:
+- Basic connectivity between nodes (i.e. nodes can all ping each other)
+- Docker installed (and monitored to be always running)
+- One of the supported OS
+
+We will provide a node specification conformance test that will verify if provisioning has been successful. 
+
+This step is provider specific and will be implemented for each cloud provider + OS distribution separately
+using provider specific technology (cloud formation, deployment manager, PXE boot, etc).
+Some OS distributions may meet the provisioning criteria without needing to run any post-boot steps as they
+ship with all of the requirements for the node specification by default.
+
+**Substeps** (on the GCE example):
+
+1. Create network
+2. Create firewall rules to allow communication inside the cluster
+3. Create firewall rule to allow ```ssh``` to all machines
+4. Create firewall rule to allow ```https``` to master
+5. Create persistent disk for master
+6. Create static IP address for master
+7. Create master machine
+8. Create node machines
+9. Install docker on all machines
+
+**Exit critera**:
+
+1. Can ```ssh``` to all machines and run a test docker image
+2. Can ```ssh``` to master and nodes and ping other machines
```

Comment:
You need to expand this to say that we can ping a container running on the docker bridge.  Since we pretty much are guaranteed to be able to ping the machines.  It's pinging the containers that run under Docker that we care about.